### ISSUE TITLE
NPE thrown when starting com.onfido.android.sdk.capture.ui.OnfidoActivity

### ISSUE THREAD

**Issue by KwunHeiChanCrypto** [onfido/onfido-android-sdk#174]
NPE thrown when starting Onfido flow, after adding `withLocale` when building `OnfidoConfig`

#### Version info:

- `onfido-capture-sdk` version: 9.3.1

- `compileSdkVersion`: 29
- `targetSdkVersion`: 29
- `minSdkVersion`: 26

#### Additional LogCat output:
```
java.lang.RuntimeException: Unable to start activity ComponentInfo{co.mona.android/com.onfido.android.sdk.capture.ui.OnfidoActivity}: java.lang.NullPointerException
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3792)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3968)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8512)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1139)

Caused by: java.lang.NullPointerException
        at java.util.Locale.<init>(Locale.java:738)
        at java.util.Locale.<init>(Locale.java:791)
        at com.google.gson.internal.bind.TypeAdapters$28.read(TypeAdapters:683)
        at com.google.gson.internal.bind.TypeAdapters$28.read(TypeAdapters:661)
        at com.google.gson.Gson.fromJson(Gson:932)
        at com.newrelic.agent.android.instrumentation.GsonInstrumentation.fromJson(GsonInstrumentation:126)
        at com.google.gson.Gson.fromJson(Gson:897)
        at com.newrelic.agent.android.instrumentation.GsonInstrumentation.fromJson(GsonInstrumentation:117)
        at com.google.gson.Gson.fromJson(Gson:846)
        at com.newrelic.agent.android.instrumentation.GsonInstrumentation.fromJson(GsonInstrumentation:99)
        at com.onfido.android.sdk.capture.common.preferences.PreferencesManager.getLocale(Unknown:26)
        at androidx.appcompat.app.l.H(Unknown:5)
        at androidx.appcompat.app.l.d(Unknown:19)
        at androidx.appcompat.app.AppCompatActivity.attachBaseContext(AppCompatActivity:139)
        at android.app.Activity.attach(Activity.java:8078)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3724)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3968)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2307)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8512)
        at java.lang.reflect.Method.invoke(Method.java:-2)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1139)
```

**Comment by gokhanaliccii** [comment]
Hi @KwunHeiChanCrypto,

Sorry for the NPE, can you please share your integration code?

**Comment by KwunHeiChanCrypto** [comment]
Hi @gokhanaliccii sorry for the late reply, here it is
```
val config = OnfidoConfig.builder(this)
    .withLocale(languageManager.getLocale())
    .withSDKToken(token, object : TokenExpirationHandler {
        override fun refreshToken(injectNewToken: (String?) -> Unit) {
            injectNewToken
        }
    })
    .withCustomFlow(arrayOf(FlowStep.WELCOME, FlowStep.CAPTURE_DOCUMENT, FlowStep.CAPTURE_FACE, FlowStep.FINAL))
    .build()
```

**Comment by gokhanaliccii** [comment]
Seems the problem occurs at the locale object's deserialization `Gson.fromJson` but I didn't manage to reproduce the NPE on the [sample app](https://github.com/onfido/onfido-android-sdk/tree/master/sample-app) when I tried different Locale values like `Locale.ENGLISH)`, `Locale.GERMAN` etc. 

We could catch the deserialization problem to prevent NPE but locale would be ignored. 

Can you please help us to find the root cause of the problem by answering new questions :)  

- Do you know if this problem occurs in a specific Locale?

- Do you get the NPE when you start the SDK with `Locale.ENGLISH`

```
val config = OnfidoConfig.builder(this)
    .withLocale(Locale.ENGLISH)
    .withSDKToken(token)
    .withCustomFlow(arrayOf(FlowStep.WELCOME, FlowStep.CAPTURE_DOCUMENT, FlowStep.CAPTURE_FACE, FlowStep.FINAL))
    .build()
```

- Do you experience any problem when you remove "withLocale" option ?

**Comment by KwunHeiChanCrypto** [comment]
1. Not really, I would have to add some more logging to get this information accurately. But the function `languageManager.getLocale()` could return `Locale.ROOT` in certain cases, would that cause the NPE?
2. I tried with `Locale.ENGLISH` a few times, no NPE
3. No, cause the NPE only occurred after adding `withLocale`

**Comment by gokhanaliccii** [comment]
Thanks @KwunHeiChanCrypto, 

I've reproduced the problem when I initialized the SDK with `.withLocale(Locale.ROOT)` I've created a ticket in our backlog once we have an update on this problem we will update you.

Can you please update your implementation to not start the SDK with `Locale.ROOT` till we fix the problem to prevent new crashes?

**Comment by KwunHeiChanCrypto** [comment]
@gokhanaliccii sure sounds great

**Comment by musagil** [comment]
Hi @KwunHeiChanCrypto ,
The issues are fixed in our new release 11.0.0, please update your dependency and let us know if you could reproduce this again.

I'm closing the issue feel free to open it if you think the issue hasn't been resolved
